auraed/src/vms/virtual_machines.rs

```diff
@@ -0,0 +1,90 @@
+use std::collections::HashMap;
+
+use anyhow::anyhow;
+use vmm_sys_util::signal::block_signal;
+
+use super::virtual_machine::{VirtualMachine, VmID, VmSpec};
+
+type Cache = HashMap<VmID, VirtualMachine>;
+
+/// The in-memory cache of virtual machines ([VirtualMachine]) created with Aurae.
+#[derive(Debug)]
+pub struct VirtualMachines {
+    cache: Cache,
+}
+
+impl VirtualMachines {
+    /// Create a new instance of the virtual machines cache.
+    pub fn new() -> Self {
+        unsafe {
+            let _ = libc::signal(libc::SIGCHLD, libc::SIG_IGN);
+        }
+
+        // Before we start any threads, mask the signals we'll be
```

Comment:
This is adapted almost directly from Cloud Hypervisor ([here](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/98d0aabb99dcdbf5e8a1e6ebc0aa1ab72b3c1114/src/main.rs#L595-L613)). I'm not quite sure whether this needs to be run before each VM is provisioned, or just at startup.